tests/integ/test_linear_learner.py

```diff
@@ -92,6 +92,36 @@ def test_linear_learner(sagemaker_session):
             assert record.label["score"] is not None
 
 
+def test_linear_learner_multiclass(sagemaker_session):
+    with timeout(minutes=15):
+        data_path = os.path.join(DATA_DIR, 'one_p_mnist', 'mnist.pkl.gz')
+        pickle_args = {} if sys.version_info.major == 2 else {'encoding': 'latin1'}
+
+        # Load the data into memory as numpy arrays
+        with gzip.open(data_path, 'rb') as f:
+            train_set, _, _ = pickle.load(f, **pickle_args)
+
+        train_set = train_set[0], train_set[1].astype(np.dtype('float32'))
+
+        ll = LinearLearner('SageMakerRole', 1, 'ml.c4.2xlarge', base_job_name='test-linear-learner',
+                           predictor_type='multiclass_classifier', sagemaker_session=sagemaker_session)
+
+        ll.epochs = 1
+        ll.num_classes = 10
```

Comment:
Our tests catched that it's too late to set num_classes here. Before this point the validation already fails on num_classes.